java/src/main/java/ai/rapids/cudf/HostColumnVector.java

```diff
@@ -738,15 +738,49 @@ public ColumnBuilder appendLists(List... inputLists) {
     public ColumnBuilder appendStructValues(List<StructData> inputList) {
       for (StructData structInput : inputList) {
         // one row
-        rows++;
         append(structInput);
       }
       return this;
     }
 
+    private void resizeDataBuffer(int length) {
+      // just for strings we want to throw a real exception if we would overrun the buffer
+      if (data == null) {
+        // since rows can be updated as we go, make data buffer as big as the estimate or the requested length
+        if (!type.isNestedType() && type != DType.STRING) {
+          data = HostMemoryBuffer.allocate(Math.min(Integer.MAX_VALUE, estimatedRows * type.getSizeInBytes()));
+        } else {
+          data = HostMemoryBuffer.allocate(length);
+        }
+        return;
+      }
+      long oldLen = data.getLength();
+      long newLen = Math.max(1, oldLen);
+      while (currentByteIndex + length > newLen) {
+        newLen *= 2;
+      }
+      if (newLen > Integer.MAX_VALUE) {
```

Comment:
Yes that is what I was referring to here :
> Agree that the index limit /bye indices would be int max. I might have to revisit it here since I may have missed the point here. To me, how do you fit ints with row count =int max in a data buffer of int max limit where every entry is say 4 bytes long. Sorry about the back and forth here, is appreciate the explanation.
>
Sounds good, I will make a change to allow that for fixed types and keep it this was for string.